networkit/cpp/io/SNAPEdgeListPartitionReader.cpp

```diff
@@ -51,7 +53,9 @@ Cover SNAPEdgeListPartitionReader::read(const std::string &path,
         } else {
             std::stringstream linestream(line);
             while (linestream >> current) {
+#ifndef NDEBUG
                 uniqueIDs.insert(current);
+#endif
```

Comment:
What does this do?

Reply:
If NDEBUG is set, it does nothing. Otherwise, it inserts the element into a set, but even if it does, there is no fail state?

Reply:
There seems to be a debug log using `uniqueIDs` below, I think letting this depend on `NDEBUG` is therefore not the best option.

Reply:
You cannot assume that `DEBUG` is only compiled in when NDEBUG is not set. Why does this pass CI?

Reply:
That's why I asked what to do. We could either add the `#ifndef NDEBUG` to the log message (so that we do not compile it in release mode), or we just remove it so that the set is not needed anymore.

Reply:
I guess the message was done for a reason, so putting it into `#ifndef` sounds reasonable.

Reply:
Agree, I put the debug message within the `#ifndef`.